From 53d803058e7a296b4a60b4bfafd33383c5786ef4 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Fri, 22 Apr 2022 15:24:41 +0000
Subject: [PATCH] libstdc++: fix incomplete type of media::MediaLog

Destructor of std::unique_ptr in libstdc++ uses sizeof() which
requires full definition of media::MediaLog in blink::Platform.
Move definition to platform.cc to avoid large header inclusion.
---
 third_party/blink/public/platform/platform.h             | 4 +---
 third_party/blink/renderer/platform/exported/platform.cc | 8 ++++++++
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/third_party/blink/public/platform/platform.h b/third_party/blink/public/platform/platform.h
index 3a8c825..3f8fbe6 100644
--- a/third_party/blink/public/platform/platform.h
+++ b/third_party/blink/public/platform/platform.h
@@ -811,9 +811,7 @@ class BLINK_PLATFORM_EXPORT Platform {
   virtual std::unique_ptr<media::MediaLog> GetMediaLog(
       MediaInspectorContext* inspector_context,
       scoped_refptr<base::SingleThreadTaskRunner> owner_task_runner,
-      bool is_on_worker) {
-    return nullptr;
-  }
+      bool is_on_worker);
 
   // GpuVideoAcceleratorFactories --------------------------------------
 
diff --git a/third_party/blink/renderer/platform/exported/platform.cc b/third_party/blink/renderer/platform/exported/platform.cc
index 496596f..b9cd638 100644
--- a/third_party/blink/renderer/platform/exported/platform.cc
+++ b/third_party/blink/renderer/platform/exported/platform.cc
@@ -40,6 +40,7 @@
 #include "build/build_config.h"
 #include "components/viz/common/gpu/raster_context_provider.h"
 #include "gpu/ipc/client/gpu_channel_host.h"
+#include "media/base/media_log.h"
 #include "services/network/public/cpp/shared_url_loader_factory.h"
 #include "third_party/blink/public/common/thread_safe_browser_interface_broker_proxy.h"
 #include "third_party/blink/public/platform/scheduler/web_thread_scheduler.h"
@@ -381,4 +382,11 @@ gfx::ColorSpace Platform::GetRenderingColorSpace() const {
   return {};
 }
 
+std::unique_ptr<media::MediaLog> Platform::GetMediaLog(
+    MediaInspectorContext* inspector_context,
+    scoped_refptr<base::SingleThreadTaskRunner> owner_task_runner,
+    bool is_on_worker) {
+  return nullptr;
+}
+
 }  // namespace blink
-- 
2.35.1

